coreneuron/nrniv/nrn_setup.cpp

```diff
@@ -2055,23 +2058,23 @@ for (int i=0; i < nt.end; ++i) {
         // Does this thread have this type.
         Memb_list* ml = nt._ml_list[type];
         if (ml) {  // needs a NetSendBuffer
-            NetSendBuffer_t* nsb = (NetSendBuffer_t*)ecalloc(1, sizeof(NetSendBuffer_t));
+            NetSendBuffer_t* nsb = (NetSendBuffer_t*)ecalloc_align(1, NRN_SOA_BYTE_ALIGN, sizeof(NetSendBuffer_t));
             ml->_net_send_buffer = nsb;
 
             // begin with a size equal to twice number of instances
             // at present there is no provision for dynamically increasing this.
             nsb->_size = ml->nodecount * 2;
             nsb->_cnt = 0;
 
-            nsb->_sendtype = (int*)ecalloc(nsb->_size, sizeof(int));
-            nsb->_vdata_index = (int*)ecalloc(nsb->_size, sizeof(int));
-            nsb->_pnt_index = (int*)ecalloc(nsb->_size, sizeof(int));
-            nsb->_weight_index = (int*)ecalloc(nsb->_size, sizeof(int));
+            nsb->_sendtype = (int*)ecalloc_align(nsb->_size, NRN_SOA_BYTE_ALIGN, sizeof(int));
+            nsb->_vdata_index = (int*)ecalloc_align(nsb->_size, NRN_SOA_BYTE_ALIGN, sizeof(int));
+            nsb->_pnt_index = (int*)ecalloc_align(nsb->_size, NRN_SOA_BYTE_ALIGN, sizeof(int));
+            nsb->_weight_index = (int*)ecalloc_align(nsb->_size, NRN_SOA_BYTE_ALIGN, sizeof(int));
             // when == 1, NetReceiveBuffer_t is newly allocated (i.e. we need to free previous copy
             // and recopy new data
             nsb->reallocated = 1;
-            nsb->_nsb_t = (double*)ecalloc(nsb->_size, sizeof(double));
-            nsb->_nsb_flag = (double*)ecalloc(nsb->_size, sizeof(double));
+            nsb->_nsb_t = (double*)ecalloc_align(nsb->_size, NRN_SOA_BYTE_ALIGN, sizeof(double));
+            nsb->_nsb_flag = (double*)ecalloc_align(nsb->_size, NRN_SOA_BYTE_ALIGN, sizeof(double));
```

Comment:
By looking at these changes, what if we we change : 

```cpp
inline void* ecalloc_align(size_t n, size_t alignment, size_t size) {
```

to 

```
inline void* ecalloc_align(size_t n,  size_t size, size_t alignment = NRN_SOA_BYTE_ALIGN) {
```

i.e. ecalloc_align should do allocations using NRN_SOA_BYTE_ALIGN alignment by default.

Reply:
Done!